### PR TITLE
[MacOS] FactSet Alignment fix

### PR DESCRIPTION
**Pull request by snbehar**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift [hunk 1]
```diff
@@ -26,6 +26,7 @@ class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
         valueStack.orientation = .vertical
         valueStack.alignment = .leading
         
+        var requiredWidth = CGFloat(0)
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift:29]:
```suggestion
        var requiredWidth: CGFloat = 0
```

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift [hunk 2]
```diff
@@ -52,6 +53,7 @@ class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
             
             if !(titleView.isEmpty) || !(valueView.isEmpty) {
                 titleStack.addArrangedSubview(titleView)
+                requiredWidth = (titleView.attributedTextValue?.size().width ?? 0) > requiredWidth ? (titleView.attributedTextValue?.size().width ?? 0) : requiredWidth
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift:56]:
```suggestion
                requiredWidth = max(titleView.attributedTextValue?.size().width ?? 0, requiredWidth)
```

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift [hunk 3]
```diff
@@ -66,7 +68,9 @@ class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
         // Spacing between title and value in the horizontal Stack
         titleStack.trailingAnchor.constraint(equalTo: valueStack.leadingAnchor, constant: -10).isActive = true
         // Getting Max width from Host config if it exists
-        titleStack.widthAnchor.constraint(lessThanOrEqualToConstant: CGFloat(truncating: factsetConfig?.title.maxWidth ?? 150)).isActive = true
+        if let maxAllowedWidth = factsetConfig?.title.maxWidth, requiredWidth > CGFloat(truncating: maxAllowedWidth) {
+            titleStack.widthAnchor.constraint(equalToConstant: CGFloat(truncating: maxAllowedWidth)).isActive = true
+        }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift:73]: Guess in the else part we have to constraint is to `requiredWidht` ?

**Review by blue-stone99 — COMMENTED** [review]
The logic looks good @snbehar. Just a couple of suggestions.
Also, there are no ambigous constraints breaking right?

**Review by mukuagar — APPROVED** [review]

**Review by rohshar6 — APPROVED** [review]

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift [hunk 4]
```diff
@@ -66,7 +68,12 @@ class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
         // Spacing between title and value in the horizontal Stack
         titleStack.trailingAnchor.constraint(equalTo: valueStack.leadingAnchor, constant: -10).isActive = true
         // Getting Max width from Host config if it exists
-        titleStack.widthAnchor.constraint(lessThanOrEqualToConstant: CGFloat(truncating: factsetConfig?.title.maxWidth ?? 150)).isActive = true
+        let maxAllowedWidth = CGFloat(truncating: factsetConfig?.title.maxWidth ?? 150)
+        if requiredWidth > maxAllowedWidth {
+            titleStack.widthAnchor.constraint(equalToConstant: CGFloat(maxAllowedWidth)).isActive = true
+        } else {
+            titleStack.widthAnchor.constraint(equalToConstant: CGFloat(requiredWidth + 2)).isActive = true
+        }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift:76]:
```suggestion
        if requiredWidth > maxAllowedWidth {
            titleStack.widthAnchor.constraint(equalToConstant: maxAllowedWidth).isActive = true
        } else {
            titleStack.widthAnchor.constraint(equalToConstant: requiredWidth + 2).isActive = true
        }
```
Explicit `CGFloat` conversion is not required.

**Pull request merged**